### PR TITLE
Handle deserialization exception and mark all test as failed

### PR DESCRIPTION
**Pull request by negiDharmendra**

### CODE REVIEW

File: Gauge.VisualStudio.TestAdapter/GaugeRunner.cs [hunk 1]
```diff
@@ -137,7 +137,17 @@ private void OnOutputDataReceived(object sender, DataReceivedEventArgs args)
...
-                var e = (TestExecutionEvent) serializer.ReadObject(ms);
+                TestExecutionEvent e;
+                try
+                {
+                    e = (TestExecutionEvent)serializer.ReadObject(ms);
+                }
+                catch (Exception err)
+                {
+                    _frameworkHandle.SendMessage(TestMessageLevel.Informational, $"Failed to deserialize : {args.Data}\n Error : {err}");
+                    MarkAllTestsFailed();
```
Comment (sriv) [inline Gauge.VisualStudio.TestAdapter/GaugeRunner.cs]: Is it expected to mark all tests failed? or the current test skipped?

Comment (negiDharmendra) [reply Gauge.VisualStudio.TestAdapter/GaugeRunner.cs]: We don't have information about the current test being executed as deserialization has failed.

File: Gauge.VisualStudio.TestAdapter/GaugeRunner.cs [hunk 2]
```diff
@@ -189,6 +200,18 @@ private void OnOutputDataReceived(object sender, DataReceivedEventArgs args)
             }
         }
 
+        private static string EscapeCharacterSequence(string data)
+        {
+            data = Regex.Replace(data, @"([^\\])\\([^\\])", "$1\\\\$2");
```
Comment (sriv) [inline Gauge.VisualStudio.TestAdapter/GaugeRunner.cs]: you don't need regex for string replace, `String.Replace` would be better, and performant.

If you need to use regex, you can have one regex for all characters to be escaped, and not one per character.

Also, if you are using regex, please use `RegexOptions.Compiled` (ref: https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=netframework-4.8#fields)

Comment (negiDharmendra) [reply Gauge.VisualStudio.TestAdapter/GaugeRunner.cs]: This whole scaping bit has been removed from `gauge-visualstudio`. Now **gauge** will take care of proper JSON encoding.

**Pull request merged**